File: src/main/java/app/application/controller/main_window/SettingsPanelController.java
package app.application.controller.main_window;

import app.application.data.VersionProperties;
import app.application.utils.DialogManager;
import app.application.utils.UserConfigHandler;
import javafx.fxml.FXML;
import javafx.scene.control.CheckBox;
import javafx.scene.control.TextField;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class SettingsPanelController {

	@FXML
	private TextField txtDownloadPath;

	@FXML
	private TextField txtVerion;

	@FXML
	private CheckBox chkOverwriteVideos;

	@Autowired
	private UserConfigHandler userConfigHandler;

	@Autowired
	private VersionProperties versionProperties;

	@Autowired
	private DialogManager dialogManager;

	@FXML
	public void initialize(){
		txtDownloadPath.textProperty().bindBidirectional(userConfigHandler.getUserConfig().getDownloadDir());
		chkOverwriteVideos.selectedProperty().bindBidirectional(userConfigHandler.getUserConfig().getOverwriteExistingVideo());
		txtVerion.textProperty().bind(versionProperties.getVersion());
	}

	public void btnSave_click(){
		userConfigHandler.writeConfig();
		dialogManager.openInformationDialog("Speichern erfolgreich", "");
	}

}

File: src/main/java/app/application/controller/main_window/SettingsPanelController.kt
package app.application.controller.main_window

import app.application.data.VersionProperties
import app.application.utils.DialogManager
import app.application.utils.UserConfigHandler
import javafx.fxml.FXML
import javafx.scene.control.CheckBox
import javafx.scene.control.TextField
import org.springframework.stereotype.Component

@Component
class SettingsPanelController(
        private val userConfigHandler: UserConfigHandler,
        private val versionProperties: VersionProperties,
        private val dialogManager: DialogManager) {

    @FXML
    private lateinit var txtDownloadPath: TextField

    @FXML
    private lateinit var txtVerion: TextField

    @FXML
    private lateinit var chkOverwriteVideos: CheckBox

    @FXML
    private lateinit var chkSubfolderForPlaylists: CheckBox

    
    @FXML
    fun initialize() {
        txtDownloadPath.textProperty().bindBidirectional(userConfigHandler.userConfig!!.downloadDir)
        chkOverwriteVideos.selectedProperty().bindBidirectional(userConfigHandler.userConfig!!.overwriteExistingVideo)
        chkSubfolderForPlaylists.selectedProperty().bindBidirectional(userConfigHandler.userConfig!!.subFolderForPlaylists)
        txtVerion.textProperty().bind(versionProperties.version)
    }

    fun btnSaveClick() {
        userConfigHandler.writeConfig()
        dialogManager.openInformationDialog("Speichern erfolgreich", "")
    }
}